From 34df3fb1249d6f38578c184e13603e0618add255 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9rgio=20M=2E=20Basto?= <sergio@serjux.com>
Date: Sun, 23 Nov 2025 04:21:10 +0000
Subject: [PATCH 1/9] New version without bundled Apache Commons v2

Add org.apache.commons:commons-text dependency to pom.xml and core/pom.xml
---
 core/pom.xml                                  |  4 ++++
 .../com/biglybt/core/metasearch/Result.java   |  4 ++--
 .../core/metasearch/impl/web/WebResult.java   | 24 +++++++++----------
 .../SimpleXMLParserDocumentImpl.java          |  8 +++----
 pom.xml                                       |  5 ++++
 5 files changed, 27 insertions(+), 18 deletions(-)

diff --git a/core/pom.xml b/core/pom.xml
index f5c891b4d..2847692f5 100644
--- a/core/pom.xml
+++ b/core/pom.xml
@@ -12,6 +12,10 @@
 	</parent>
 
 	<dependencies>
+		<dependency>
+			<groupId>org.apache.commons</groupId>
+			<artifactId>commons-text</artifactId>
+		</dependency>
 		<dependency>
 			<groupId>commons-cli</groupId>
 			<artifactId>commons-cli</artifactId>
diff --git a/core/src/com/biglybt/core/metasearch/Result.java b/core/src/com/biglybt/core/metasearch/Result.java
index 1d7219a63..0b9c3734e 100644
--- a/core/src/com/biglybt/core/metasearch/Result.java
+++ b/core/src/com/biglybt/core/metasearch/Result.java
@@ -23,7 +23,7 @@ import java.net.InetAddress;
 import java.net.URL;
 import java.util.*;
 
-import org.apache.commons.lang.Entities;
+import org.apache.commons.text.StringEscapeUtils;
 import org.json.simple.JSONObject;
 
 import com.biglybt.core.metasearch.utils.MomentsAgoDateFormatter;
@@ -512,6 +512,6 @@ Result
 		if ( input == null ){
 			return( null );
 		}
-		return( Entities.HTML40.unescape( input ));
+		return( StringEscapeUtils.unescapeHtml4( input ));
 	}
 }
diff --git a/core/src/com/biglybt/core/metasearch/impl/web/WebResult.java b/core/src/com/biglybt/core/metasearch/impl/web/WebResult.java
index 67a80bca7..83114c975 100644
--- a/core/src/com/biglybt/core/metasearch/impl/web/WebResult.java
+++ b/core/src/com/biglybt/core/metasearch/impl/web/WebResult.java
@@ -21,7 +21,7 @@ package com.biglybt.core.metasearch.impl.web;
 
 import java.util.*;
 
-import org.apache.commons.lang.Entities;
+import org.apache.commons.text.StringEscapeUtils;
 
 import com.biglybt.core.metasearch.Engine;
 import com.biglybt.core.metasearch.Result;
@@ -93,14 +93,14 @@ public class WebResult extends Result {
 	public void setNameFromHTML(String name) {
 		if(name != null) {
 			name = removeHTMLTags(name);
-			this.name = Entities.HTML40.unescape(name);
+			this.name = StringEscapeUtils.unescapeHtml4(name);
 		}
 	}
 
 	public void setCommentsFromHTML(String comments) {
 		if(comments != null) {
 			comments = removeHTMLTags(comments);
-			comments = Entities.HTML40.unescape(comments);
+			comments = StringEscapeUtils.unescapeHtml4(comments);
 			comments = comments.replaceAll(",", "");
 			comments = comments.replaceAll(" ", "");
 			try{
@@ -122,7 +122,7 @@ public class WebResult extends Result {
 		if(category != null) {
 			addTagFromHTML(category);	// hack, we currently just pick up cats
 			category = removeHTMLTags(category);
-			this.category = Entities.HTML40.unescape(category).trim();
+			this.category = StringEscapeUtils.unescapeHtml4(category).trim();
 			/*int separator = this.category.indexOf(">");
 
 			if(separator != -1) {
@@ -144,7 +144,7 @@ public class WebResult extends Result {
 		}
 		
 		tag = removeHTMLTags( tag );
-		tag = Entities.HTML40.unescape( tag ).trim();
+		tag = StringEscapeUtils.unescapeHtml4( tag ).trim();
 		
 		if ( !tag.isEmpty()){
 			
@@ -173,7 +173,7 @@ public class WebResult extends Result {
 	public void setNbPeersFromHTML(String nbPeers) {
 		if(nbPeers != null) {
 			nbPeers = removeHTMLTags(nbPeers);
-			String nbPeersS = Entities.HTML40.unescape(nbPeers);
+			String nbPeersS = StringEscapeUtils.unescapeHtml4(nbPeers);
 			nbPeersS = nbPeersS.replaceAll(",", "");
 			nbPeersS = nbPeersS.replaceAll(" ", "");
 			try {
@@ -188,7 +188,7 @@ public class WebResult extends Result {
 	public void setNbSeedsFromHTML(String nbSeeds) {
 		if(nbSeeds != null) {
 			nbSeeds = removeHTMLTags(nbSeeds);
-			String nbSeedsS = Entities.HTML40.unescape(nbSeeds);
+			String nbSeedsS = StringEscapeUtils.unescapeHtml4(nbSeeds);
 			nbSeedsS = nbSeedsS.replaceAll(",", "");
 			nbSeedsS = nbSeedsS.replaceAll(" ", "");
 			try {
@@ -203,7 +203,7 @@ public class WebResult extends Result {
 	public void setNbSuperSeedsFromHTML(String nbSuperSeeds) {
 		if(nbSuperSeeds != null) {
 			nbSuperSeeds = removeHTMLTags(nbSuperSeeds);
-			String nbSuperSeedsS = Entities.HTML40.unescape(nbSuperSeeds);
+			String nbSuperSeedsS = StringEscapeUtils.unescapeHtml4(nbSuperSeeds);
 			nbSuperSeedsS = nbSuperSeedsS.replaceAll(",", "");
 			nbSuperSeedsS = nbSuperSeedsS.replaceAll(" ", "");
 			try {
@@ -289,7 +289,7 @@ public class WebResult extends Result {
 	public void setPublishedDateFromHTML(String publishedDate) {
 		if(publishedDate != null && publishedDate.length() > 0) {
 			publishedDate = removeHTMLTags(publishedDate);
-			String publishedDateS = Entities.HTML40.unescape(publishedDate).replace((char)160,(char)32);
+			String publishedDateS = StringEscapeUtils.unescapeHtml4(publishedDate).replace((char)160,(char)32);
 			this.publishedDate = dateParser.parseDate(publishedDateS);
 		}
 	}
@@ -305,7 +305,7 @@ public class WebResult extends Result {
 	public void setSizeFromHTML(String size, long minAcceptable) {
 		if(size != null) {
 			size = removeHTMLTags(size);
-			String sizeS = Entities.HTML40.unescape(size).replace((char)160,(char)32);
+			String sizeS = StringEscapeUtils.unescapeHtml4(size).replace((char)160,(char)32);
 			sizeS = sizeS.replaceAll("<[^>]+>", " ");
 			//Add a space between the digits and unit if there is none
 			sizeS = sizeS.replaceFirst("(\\d)([a-zA-Z])", "$1 $2");
@@ -338,7 +338,7 @@ public class WebResult extends Result {
 	public void setVotesFromHTML(String votes_str) {
 		if(votes_str != null) {
 			votes_str = removeHTMLTags(votes_str);
-			votes_str = Entities.HTML40.unescape(votes_str);
+			votes_str = StringEscapeUtils.unescapeHtml4(votes_str);
 			votes_str = votes_str.replaceAll(",", "");
 			votes_str = votes_str.replaceAll(" ", "");
 			try {
@@ -352,7 +352,7 @@ public class WebResult extends Result {
 	public void setVotesDownFromHTML(String votes_str) {
 		if(votes_str != null) {
 			votes_str = removeHTMLTags(votes_str);
-			votes_str = Entities.HTML40.unescape(votes_str);
+			votes_str = StringEscapeUtils.unescapeHtml4(votes_str);
 			votes_str = votes_str.replaceAll(",", "");
 			votes_str = votes_str.replaceAll(" ", "");
 			try {
diff --git a/core/src/com/biglybt/pifimpl/local/utils/xml/simpleparser/SimpleXMLParserDocumentImpl.java b/core/src/com/biglybt/pifimpl/local/utils/xml/simpleparser/SimpleXMLParserDocumentImpl.java
index 86e6744ed..48eede362 100644
--- a/core/src/com/biglybt/pifimpl/local/utils/xml/simpleparser/SimpleXMLParserDocumentImpl.java
+++ b/core/src/com/biglybt/pifimpl/local/utils/xml/simpleparser/SimpleXMLParserDocumentImpl.java
@@ -31,7 +31,7 @@ import java.util.Vector;
 import javax.xml.parsers.DocumentBuilder;
 import javax.xml.parsers.DocumentBuilderFactory;
 
-import org.apache.commons.lang.Entities;
+import org.apache.commons.text.StringEscapeUtils;
 import org.w3c.dom.Document;
 import org.w3c.dom.Node;
 import org.xml.sax.*;
@@ -748,11 +748,11 @@ SimpleXMLParserDocumentImpl
 
 		     				}else{
 
-			     				int num = Entities.HTML40.entityValue( ref );
+			     				String decoded = StringEscapeUtils.unescapeHtml4("&" + ref + ";");
 
-		     					if ( num != -1 ){
+		     					if ( decoded != null && !decoded.equals("&" + ref + ";") ){
 
-		     						replacement = "&#" + num + ";";
+		     						replacement = decoded;
 
 		     					}else{
 
diff --git a/pom.xml b/pom.xml
index f77e797be..10f282da1 100644
--- a/pom.xml
+++ b/pom.xml
@@ -62,6 +62,11 @@
 				<artifactId>commons-cli</artifactId>
 				<version>1.4</version>
 			</dependency>
+			<dependency>
+				<groupId>org.apache.commons</groupId>
+				<artifactId>commons-text</artifactId>
+				<version>1.10.0</version>
+			</dependency>
 		</dependencies>
 	</dependencyManagement>
 
-- 
2.51.1

